### PR TITLE
chapter_12

### PR DESCRIPTION
**Pull request by FirdausIsa**
ready

### CODE REVIEW

**Comment by FirdausIsa** [comment]
ready

**Comment by FirdausIsa** [comment]
ready

**Comment by FirdausIsa** [comment]
ready

File: app/assets/stylesheets/password_resets.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the PasswordResets controller here.
```
Comment (huongnt-2545) [inline app/assets/stylesheets/password_resets.scss:1]: delete this file

File: app/helpers/password_resets_helper.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+module PasswordResetsHelper
```
Comment (huongnt-2545) [inline app/helpers/password_resets_helper.rb:1]: delete this file

File: app/controllers/password_resets_controller.rb [hunk 3]
```diff
@@ -0,0 +1,65 @@
+class PasswordResetsController < ApplicationController
+  before_action :get_user, only: [:edit, :update]
+  before_action :valid_user, only: [:edit, :update]
+  before_action :check_expiration, only: [:edit, :update]
```
Comment (huongnt-2545) [inline app/controllers/password_resets_controller.rb:4]: line 2, 3, 4 =>
```suggestion
  before_action :get_user, :valid_user, :check_expiration, only: [:edit, :update]
```

File: config/routes.rb [hunk 4]
```diff
@@ -1,4 +1,6 @@
 Rails.application.routes.draw do
+  get 'password_resets/new'
```
Comment (huongnt-2545) [inline config/routes.rb:2]: remove line 2, 3

File: test/controllers/password_resets_controller_test.rb [hunk 5]
```diff
@@ -0,0 +1,13 @@
+require "test_helper"
```
Comment (huongnt-2545) [inline test/controllers/password_resets_controller_test.rb:1]: delete this file

**Comment by FirdausIsa** [comment]
ready

File: app/controllers/password_resets_controller.rb [hunk 6]
```diff
@@ -0,0 +1,63 @@
+class PasswordResetsController < ApplicationController
+  before_action :get_user, :valid_user, :check_expiration, only: [:edit, :update]
+
+  def new
+  end
+
+  def edit
+  end
```
Comment (anlvq-1058) [inline app/controllers/password_resets_controller.rb:8]:
```suggestion
  def new; end

  def edit; end
```

File: app/views/password_resets/edit.html.erb [hunk 7]
```diff
@@ -0,0 +1,17 @@
+<% provide(:title, "Reset password") %>
+<h1>Reset password</h1>
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for(@user, url: password_reset_path(params[:id])) do |f| %>
+      <%= render 'shared/error_messages' %>
+      <%= hidden_field_tag :email, @user.email%>
```
Comment (anlvq-1058) [inline app/views/password_resets/edit.html.erb:7]:
```suggestion
      <%= hidden_field_tag :email, @user.email %>
```

File: config/webpack/environment.js [hunk 8]
```diff
@@ -1,11 +1,3 @@
 const { environment } = require('@rails/webpacker')
 
-const webpack = require('webpack')
-environment.plugins.prepend('Provide',
-  new webpack.ProvidePlugin({
-    $: 'jquery/src/jquery',
-    jQuery: 'jquery/src/jquery'
-  })
-)
-
```
Comment (anlvq-1058) [inline config/webpack/environment.js:10]: why you remove these lines of code?

**Review by anlvq-1058 — COMMENTED** [review]
cmt

**Comment by FirdausIsa** [comment]
ready

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**